guides/theming-your-components.md

```diff
@@ -1,135 +1,224 @@
-# Theming your custom component with Angular Material's theming system
+# Theme your own components with Angular Material's theming system
 
-In order to style your own components with Angular Material's tooling, the component's styles must
-be defined with Sass.
+You can use Angular Material's Sass-based theming system for your own custom components.
 
-## 1. Define all color and typography styles in a "theme file" for the component
+## Reading style values from a theme
 
-First, create a Sass mixin that accepts an Angular Material color configuration and
-outputs the color-specific styles for the component. A color configuration is a Sass map.
+As described in the [theming guide][theme-map], a theme is a Sass map that contains style values to
+customize components. Angular Material provides APIs for reading values from this data structure.
+
+[theme-map]: https://material.angular.io/guide/theming#themes
+
+### Reading color values
+
+To read color values from a theme, you can use the `get-color-config` Sass function. This function
+returns a Sass map containing the theme's primary, accent, and warn palettes, as well as a flag
+indicating whether dark mode is set.
 
-For example, if building a custom carousel component:
 ```scss
-// Import library functions for theme creation.
-@import '~@angular/material/theming';
-
-@mixin candy-carousel-color($config-or-theme) {
-  // Extract the color configuration in case a theme has been passed.
-  // This allows consumers to either pass a theme object or a color configuration.
-  $config: mat-get-color-config($config-or-theme);
-  // Extract the palettes you need from the theme definition.
-  $primary: map-get($config, primary);
-  $accent: map-get($config, accent);
-  
-  // Define any styles affected by the theme.
-  .candy-carousel {
-    // Use mat-color to extract individual colors from a palette.
-    background-color: mat-color($primary);
-    border-color: mat-color($accent, A400);
-  }
-}
+@use 'sass:map';
+@use '~@angular/material' as mat;
+
+$color-config:    mat.get-color-config($theme);
+$primary-palette: map.get($color-config, 'primary');
+$accent-palette:  map.get($color-config, 'accent');
+$warn-palette:    map.get($color-config, 'warn');
```

Comment:
I think (maybe no longer?) you can also get background and foreground so that you could style a raised carousel to have a background color like a 'card' etc? That seems useful to incorporate here for components that have background or foreground (especially with dark modes)